Source/Blazorise/Utilities/IO/RemoteFileEntryStream.cs

```diff
@@ -93,13 +94,15 @@ private async Task FillBuffer( PipeWriter writer, CancellationToken cancellation
                     catch ( Exception e )
                     {
                         await writer.CompleteAsync( e );
-                        return;
                     }
                 }
             }
             finally
             {
-                await fileEntryNotifier.UpdateFileEndedAsync( fileEntry, offset == fileEntry.Size );
+                var success = position == fileEntry.Size;
+                var fileInvalidReason = success ? FileInvalidReason.None : FileInvalidReason.UnexpectedError;
```

Comment:
Does it makes sense to also check `position != fileEntry.Size`? So that it fails with `FileInvalidReason.MaxLengthExceeded`?

Reply:
Doesn't the existing check above that exits the method early, already do that?

Reply:
The one inside of the `while` loop would only work until the end of the file is reached. If for any reason it goes over that it would need to report `MaxLengthExceeded`.

Reply:
However unlikely :)

Reply:
Ok we can add another check... actually it should check only ocne at the begin, don't you agree?

Reply:
At the beginning and at the end. Although the one at the end might be the more important.